File: src/structural/bridge.java
public abstract class Pizza {
    protected String sauce;
    protected String toppings;
    protected String crust;

    public abstract void deliver();
}

public class PepperoniPizza extends Pizza {
    @Override
    public void deliver() {
        System.out.println("Adding Sauce");
        System.out.println("Adding Toppings");
        System.out.println("Adding Pepperoni");
        System.out.println("Order in Progress!");
    }
}

public class VeggiePizza extends Pizza {
    @Override
    public void deliver() {
        System.out.println("Adding Sauce");
        System.out.println("Adding Different Toppings");
        System.out.println("Adding Cheese");
        System.out.println("Order in Progress!");
    }
}

File: src/structural/bridge.ts
abstract class Pizza {
  constructor(
    protected sauce: string,
    protected toppings: string,
    protected crust: string,
  ) {}

  abstract deliver(): void;
}

class PepperoniPizza extends Pizza {
  deliver(): void {
    console.log("Adding Sauce");
    console.log("Adding Toppings");
    console.log("Adding Pepperoni");
    console.log("Order in Progress!");
  }
}

class VeggiePizza extends Pizza {
  deliver(): void {
    console.log("Adding Sauce");
    console.log("Adding Different Toppings");
    console.log("Adding Cheese");
    console.log("Order in Progress!");
  }
}
